spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarOperations.java

```diff
@@ -103,6 +103,13 @@
 		 */
 		SendMessageBuilder<T> withCustomRouter(MessageRouter messageRouter);
 
+		/**
+		 * Specifies the customizer to use to further configure the producer builder.
+		 * @param producerCustomizer the producer builder customizer
+		 * @return the current builder with the producer builder customizer specified
+		 */
+		SendMessageBuilder<T> withProducerCustomizer(ProducerBuilderCustomizer<T> producerCustomizer);
```

Comment:
NOTE: The underlying producer factory support N for extension later but we only expose 1 to user for now for simplicity.